### PR TITLE
added button to show completed story modal if user accidentally clicks out of it 

### PR DESCRIPTION
**Pull request by AAloshine-scottlogic**
## Description
there is a congratulations modal that pops up when users complete all 3 levels. This ticket includes a button in the mission info banner to bring the win modal back up if the user clicks out of it accidentally.

## Screenshots

image

## Checklist

Have you done the following?

- [x] Linked the relevant Issue
https://github.com/ScottLogic/prompt-injection/issues/805
- [ ] Added tests
- [ ] Ensured the workflow steps are passing

### CODE REVIEW

**Comment by pmarsh-scottlogic** [comment]
protip, the answers to the levels are in the backend's _email.ts_ in `checkLevelWinCondition`
Or you can go into the browser's dev console, find the localStorage and change the value of `numCompletedLevels` to anything greater than 4.

File: frontend/src/components/LevelMissionInfoBanner/LevelMissionInfoBanner.tsx [hunk 1]
```diff
@@ -7,10 +7,16 @@ import './LevelMissionInfoBanner.css';
...
 	openOverlay,
+	openLevelsCompleteOverlay,
+	numCompletedLevels,
 }: {
 	currentLevel: LEVEL_NAMES;
 	openOverlay: () => void;
+	openLevelsCompleteOverlay: () => void;
+	numCompletedLevels: number;
 }) {
+	const isLevelComplete = (currentLevel as number) < numCompletedLevels;
```
Comment (pmarsh-scottlogic) [inline frontend/src/components/LevelMissionInfoBanner/LevelMissionInfoBanner.tsx:18]: you shouldn't need to cast here. LEVEL_NAMES is already a numeric type.

```suggestion
	const isLevelComplete = currentLevel < numCompletedLevels;
```

Comment (pmarsh-scottlogic) [reply frontend/src/components/LevelMissionInfoBanner/LevelMissionInfoBanner.tsx:18]: I would even add
```typescript
const isLevel3 = currentLevel === LEVEL_NAMES.LEVEL_3
```
and incorporate that into the conditional render below, for brevity and explicitness in the render function

**Review by pmarsh-scottlogic — CHANGES_REQUESTED** [review]
lovely stuff. Just a minor tweak

File: frontend/src/components/LevelMissionInfoBanner/LevelMissionInfoBanner.tsx (hunk 1 above)
Comment (AAloshine-scottlogic) [reply frontend/src/components/LevelMissionInfoBanner/LevelMissionInfoBanner.tsx:18]: yeah that makes sense

**Review by pmarsh-scottlogic — APPROVED** [review]
🔥

File: frontend/src/components/LevelMissionInfoBanner/LevelMissionInfoBanner.test.tsx [hunk 2]
```diff
@@ -41,6 +43,8 @@ describe('LevelMissionInfoBanner component tests', () => {
 			<LevelMissionInfoBanner
 				currentLevel={currentLevel}
 				openOverlay={openOverlayMock}
+				openLevelsCompleteOverlay={() => {}}
```
Comment (chriswilty) [inline frontend/src/components/LevelMissionInfoBanner/LevelMissionInfoBanner.test.tsx:46]: I suppose for the sake of completion, you could test that the openLevelsCompleteOverlay callback is fired on click, similar to the openOverlayMock. But there's not much to gain from it to be honest, there are bigger fish to fry.

File: frontend/src/components/LevelMissionInfoBanner/LevelMissionInfoBanner.tsx [hunk 3]
```diff
@@ -25,6 +32,11 @@ function LevelMissionInfoBanner({
 				</span>
 				Mission Info
 			</ThemedButton>
+			{isLevelComplete && isLevel3 && (
+				<ThemedButton onClick={openLevelsCompleteOverlay}>
+					Congratulations
```
Comment (chriswilty) [inline frontend/src/components/LevelMissionInfoBanner/LevelMissionInfoBanner.tsx:37]: Maybe a `!` would be nice? As in, "Congratulations!"

**Review by chriswilty — COMMENTED** [review]
Very happy with this, just a couple of small suggestions.

File: frontend/src/components/LevelMissionInfoBanner/LevelMissionInfoBanner.tsx (hunk 3 above)
Comment (AAloshine-scottlogic) [reply frontend/src/components/LevelMissionInfoBanner/LevelMissionInfoBanner.tsx:37]: sounds good

File: frontend/src/components/LevelMissionInfoBanner/LevelMissionInfoBanner.test.tsx (hunk 2 above)
Comment (AAloshine-scottlogic) [reply frontend/src/components/LevelMissionInfoBanner/LevelMissionInfoBanner.test.tsx:46]: it'll take two seconds to put in the test no problem

**Pull request merged**